generators/php/model/src/generateTraits.ts

```diff
@@ -0,0 +1,34 @@
+import { ModelGeneratorContext } from "./ModelGeneratorContext";
+import { TraitGenerator } from "./trait/TraitGenerator";
+
+export function generateTraits(context: ModelGeneratorContext): void {
+    const extendedTypeIdsFromTypes: Set<string> = new Set(
+        Object.values(context.ir.types).flatMap((typeDeclaration) =>
+            typeDeclaration.shape._visit<string[]>({
+                alias: () => [],
+                enum: () => [],
+                object: (objectDeclaration) =>
+                    objectDeclaration.extends.map((declaredTypeName) => declaredTypeName.typeId),
+                undiscriminatedUnion: () => [],
+                union: () => [],
+                _other: () => []
+            })
+        )
+    );
+    const extendedTypeIdsFromInlinedRequests = Object.values(context.ir.services)
+        .flatMap((service) => service.endpoints)
+        .flatMap((endpoint) => {
+            return endpoint.requestBody?.type === "inlinedRequestBody"
+                ? endpoint.requestBody.extends.map((declaredTypeName) => declaredTypeName.typeId)
+                : [];
+        });
+    for (const typeId of [...extendedTypeIdsFromTypes, ...extendedTypeIdsFromInlinedRequests]) {
+        const typeDeclaration = context.getTypeDeclarationOrThrow(typeId);
+        const objectTypeDeclaration = context.getUnderlyingObjectTypeDeclarationFromTypeDeclaration(typeDeclaration);
+        if (objectTypeDeclaration == null) {
+            throw new Error("Unexpected no object type declaration");
+        }
```

Comment:
nit: We could roll out a separate `getUnderlyingObjectTypeDeclarationOrThrow(typeId)` function to handle this block so the call-site reads a little simpler.

Reply:
i modified the existing methods to throw instead! let me know if you'd still like changes

i do need the `typeDeclaration` in the `TraitGenerator`, so the method still takes in a `typeDeclaration`